jcl/src/java.base/share/classes/java/lang/invoke/VarargsCollectorHandle.java

```diff
@@ -99,13 +101,39 @@ static MethodType varargsCollectorType(MethodType nextType, Class<?> arrayType)
 	}
 	
 	public Object invokeWithArguments(Object... args) throws Throwable, WrongMethodTypeException, ClassCastException {
-		if (args.length < 253) {
-			MethodHandle mh = IWAContainer.getMH(args.length);
-			return mh.invokeExact((MethodHandle)this, args);
+		int argsLength = 0;
+		if (args != null) {
+			argsLength = args.length;
+		}
+		if (argsLength < 253) {
+			MethodHandle mh = IWAContainer.getMH(argsLength);
+			return mh.invokeExact((MethodHandle) this, args); 
+		/*[IF Java10]*/
+		} else if (argsLength < 255) {
+		/*[ELSE]
+		} else {
+		/*[ENDIF]*/
+			MethodHandle mh = this.asType(MethodType.genericMethodType(argsLength));
+			mh = mh.asSpreader(Object[].class, argsLength);
+			return mh.invokeExact(args);
+		/*[IF Java10]*/
+		} else {
+			/* special handling is required if arg count exceeds the allowed number of arguments 
+			 * (254 + trailing array in this case). The additional arguments will be
+			 * stored in the trailing array and the MethodHandle can be treated as fixed arity.
+			 */
+			int mhLength = type.parameterCount();
+			int numTrailingArgs = argsLength - mhLength + 1;
+			Object trailingArgs = Array.newInstance(this.arrayType.getComponentType(), numTrailingArgs);
+			MethodHandle arraySetter = MethodHandles.arrayElementSetter(this.arrayType);
+			for (int i = 0; i < numTrailingArgs; i++) {
+				arraySetter.invoke(trailingArgs, i, args[mhLength - 1 + i]);
```

Comment:
This seems like an expensive way to set the args.  Is there a reason this has to be done with `invoke`?

Reply:
By using invoke the arguments are cast to the arrayType using "asType conversions", or the proper error is thrown if they cant be.